File: frontend/services/ProfileService.ts
import { AvatarPresignRequest } from "@/interfaces/AvatarPresignRequest";
import { axiosInstance, ENDPOINTS } from "@/lib/utils";
import { UUID } from "crypto";

export function getCurrentProfile() {
    return axiosInstance.get(ENDPOINTS.PROFILE.GET_PROFILE)
    .then(response => response.data);
}

export function updateProfile(profileData: {
    fullName?: string;
    nickName?: string;
    bio?: string;
    phoneNumber?: string;
    address?: string;
    profilePic?: string;
    id: UUID;
}) {
    return axiosInstance.put(ENDPOINTS.PROFILE.UPDATE_PROFILE, profileData)
    .then(response => response.data);
}
export function getProfileDetail(profileId: UUID) {
    return axiosInstance.get(ENDPOINTS.PROFILE.GET_PROFILE_DETAIL(profileId))
    .then(response => response.data);
}

export function getProfiles(pageNo: number, size: number, sortBy: string, ascending: boolean) {
    return axiosInstance.get(ENDPOINTS.PROFILE.GET_PROFILES, {
        params: { pageNo, size, sortBy, ascending }
    }).then(res => res.data);
}

export function searchProfiles(name: string) {
    return axiosInstance.get(ENDPOINTS.PROFILE.SEARCH_PROFILE, {
        params: { name }
    }).then(res => res.data);
}

export function followProfile(followerId: UUID) {
    return axiosInstance.post(ENDPOINTS.PROFILE.FOLLOW_PROFILE(followerId))
    .then(response => response.data);
}

export function getAvatarPresignUploadUrl(presignRequest: AvatarPresignRequest) {
    return axiosInstance.post(ENDPOINTS.PROFILE.GET_PRESIGN_UPLOAD_URL, presignRequest)
        .then(res => res.data);
}

export function uploadAvatarToS3(presignUrl: string, file: File) {

    return axiosInstance.put(presignUrl, file, {
        headers: {
            'Content-Type': file.type,
        }
    });
}

File: backend/service-profile/src/main/java/hoangworthy/project/microservices/serviceprofile/services/interfaces/ProfileService.java
package hoangworthy.project.microservices.serviceprofile.services.interfaces;


import hoangworthy.project.microservices.serviceprofile.dtos.ProfileDtoResponse;
import hoangworthy.project.microservices.serviceprofile.dtos.ProfileResponse;

import java.util.List;
import java.util.UUID;

public interface ProfileService {

    public ProfileResponse getCurrentProfile(UUID accountId);

    public List<ProfileDtoResponse> getProfiles(int pageNo, int size, String sortBy, boolean ascending);

    public List<ProfileDtoResponse> searchProfiles(String name);

    public void followProfile(UUID profileId, UUID followerId);

    public ProfileResponse getProfileInfo(UUID accountId);

    public ProfileResponse updateProfile(ProfileDtoResponse profileDtoResponse);
}
